salt/client/__init__.py

```diff
@@ -1493,10 +1493,6 @@ def pub(self,
                                                  master_uri=master_uri)
 
         try:
-            # Ensure that the event subscriber is connected.
-            # If not, we won't get a response, so error out
-            if not self.event.connect_pub(timeout=timeout):
-                raise SaltReqTimeoutError()
```

Comment:
It seems like the intent of this code is to make sure the salt-master is running before invoking `channel.send`. In theory, it seems that `channel.send` should itself timeout if the salt-master is not running, so it looks like this change should be harmless.

However, a deeper look tells me there would be a slight difference in behavior. The intent of the code may be to give the master a little bit of time to come up (due to the pausing inherent in self.event.connect_pub) so that `channel.send` will work in cases where things are slightly out-of-sync coming up.

If the master is not up at the exact time `channel.send` is called, `channel.send` would always fail with a timeout even if the master came up slightly afterwards (and still within the timeout period) because the sent packet is lost (at least in a connection-less transport like ZMQ). However, with the original code intact, as long as the master came up within the timeout period, no errors would occur.

Removing this may make such test cases slightly more unreliable.

So I would say try to remove this and see what happens. If you see more occasional test case failures, then this would be the explanation for that.